### PR TITLE
feat: Add storybook 7 compability

### PR DESCRIPTION
**Pull request by bryanjtc**
Fixes: #112

- Upgrades yarn v1 -> v3.
- Dependency upgrades: Typescript 3 -> 4, Storybook 6 - > 7, @xstate/react 1 -> 3, jest 26 ->29, eslint 7 -> 8, cypress 4 -> 12.
- Upgrades the rest of the dependencies to latest minor release.
- Upgrades github actions to latest release.
- Replaces cypress github action with script that works without issues from `wait-on`.
- Adds storybook 7 esm support.
- Moves examples to their own directory outside packages/storybook-addon-performance for better cypress testing.
- Updates cypress tests to test esm storybook config.
- Removes node 12 support and adds support for 14, 16, and 18 only. (subject to change since storybook 7 is going to drop 14)
- Adds simple react 18 support. (It just adds the range to the peer dependencies and test react 18 with cypress)
- Bundles code and dependencies with [tsup](https://github.com/egoist/tsup). 
- Replaces [gzip-js](https://github.com/beatgammit/gzip-js) with [fflate](https://github.com/101arrowz/fflate).
- Other minor fixes.

TODO: Add complete react 18 support. Depends on #117

### CODE REVIEW

File: packages/storybook-addon-performance/register.mjs [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Separate entry point for storybook addon registration
+import './dist/mjs/register';
```
Comment (DarkPurple141) [inline packages/storybook-addon-performance/register.mjs:2]: Where does this path get generated?

**Review by DarkPurple141 — APPROVED** [review]
Just one query to check otherwise LGTM

File: packages/storybook-addon-performance/register.mjs (hunk 1 above)
Comment (bryanjtc) [reply packages/storybook-addon-performance/register.mjs:2]: Line 4: "outDir": "dist/esm" of tsconfig.json inside  packages/storybook-addon-performance.

Comment (bryanjtc) [reply packages/storybook-addon-performance/register.mjs:2]: When you run typescript:build two directories are created inside dist: esm and cjs.

Comment (DarkPurple141) [reply packages/storybook-addon-performance/register.mjs:2]: This path says `mjs` tho is that a typo?

Comment (bryanjtc) [reply packages/storybook-addon-performance/register.mjs:2]: You're right.

Comment (bryanjtc) [reply packages/storybook-addon-performance/register.mjs]: Fixed it. 👍🏽

**Comment by DarkPurple141** [comment]
Just running the builds let's see what happens. Although at this stage there are no active tests verifying the esm config change.

**Comment by bryanjtc** [comment]
Not sure why the jobs fail on netlify. Storybook v7 is now on beta. I can try to expand the pr and add some tests for projects that use esm

**Comment by jgodi** [comment]
@bryanjtc  Excitingly waiting this PR -- is there anything that I can do to help?

I just discovered this addon and would love to use it. If there are beta versions of this published, I am happy to test it out.

**Comment by bryanjtc** [comment]
@DarkPurple141 Can you update the image in heroku? It's the only check failing.

File: .yarn/sdks/integrations.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# This file is automatically generated by @yarnpkg/sdks.
```
Comment (DarkPurple141) [inline .yarn/sdks/integrations.yml:1]: Are these `.yarn` files are intended to be checked in?

**Review by DarkPurple141 — APPROVED** [review]

**Comment by bryanjtc** [comment]
I will do a follow-up pr when storybook starts releasing rc for v7.

**Comment by jgodi** [comment]
Could you cut a release with this change please?

**Pull request merged**